sound/soc/sof/intel/hda-dai-ops.c

```diff
@@ -416,10 +416,12 @@ static int hda_ipc4_post_trigger(struct snd_sof_dev *sdev, struct snd_soc_dai *c
 	case SNDRV_PCM_TRIGGER_SUSPEND:
 	case SNDRV_PCM_TRIGGER_STOP:
 		/*
-		 * STOP/SUSPEND trigger is invoked only once when all users of this pipeline have
-		 * been stopped. So, clear the started_count so that the pipeline can be reset
+		 * STOP/SUSPEND trigger is invoked only once when all users of
+		 * this pipeline have been stopped. So, clear the started and
+		 * paused count so that the pipeline can be reset
 		 */
 		swidget->spipe->started_count = 0;
+		swidget->spipe->paused_count = 0;
```

Comment:
we cannot keep the streams paused, we don't support RESUME trigger, so all streams would be restarted anyways, paused or not paused.
In practice: on resume the audio will remain paused (will not run), but when you PAUSE_RELEASE it, then we will restart it. We need to re-initialize everything to get working audio.

Reply:
how will the audio remain paused if the pause_count remains zero? I don't understand how this counter is used.

Reply:
This counter is used internally to ipc4 (to track the individual pipeline states) , ALSA keeps track of the PCM state. On suspend we need to stop everything, so we can put the DSP to off.
So, when we suspend all pipelines will be reset.

Reply:
@ujfalusi this is too confusing. We increment the paused_count when we pause the stream. So lets assume this sequence:

1. Start the stream : started_count = 1
2. Pause the stream : paused_count  = 1
3. Suspend the system: The stream doesnt get the suspend trigger, so the started/paused_counts remain at 1
4. hda_dai_suspend() gets invoked so both counts get reset to 0.
5. Resume the system
6. Pause_release the stream: this will decrement the paused_count to -1?
Am I understanding this correctly?